### DISCUSSION TITLE
CHIRIMEN コミュニティの今後の活動について

### DISCUSSION THREAD

**Discussion by gurezo** [chirimen-oh/meeting#32] [General]
## 目的

- 活動目的をコミュニティ全体で、揃える事が目的です。
- cf.[CHIRIMEN コミュニティの今後の活動について #30](https://github.com/chirimen-oh/meeting/issues/30)

## ゴール

- 明確な定義が出来たら、サイトやリポジトリに活動目的を明記する

### 活動目的について

- ウェブアプリケーションからデバイス・ハードを動かしたい
- JS の API の実装のひとつ
- Web の技術で、ハードウェアを動かす
  - Web の標準技術を使う
  - JavaScript を使う
  - 素の javascript 、標準 javascript で、使える事に意味がある 
  - ライブラリ的な使い方ができる

### ユースケースを再認識

- メーカーズチャレンジ以外で、chirimen を使われているのか？ 
  - 大学や専門学校でのプログラミング授業で、活用実績あり

### 課題
- コミュニティとしての純粋な活動が減っているかも？
- エンジニアが楽しい事を企画できないか？
- リソースが不足している

### 定義について
- Web の技術で、ハードウェアを動かす
    - Web の標準技術を使う
    - JavaScript を使う
- 上記の技術を使って、プロトタイピング環境を提供する

#### プロトタイピング開発
- 誰の為に：プログラミング初学者
- 解決目的：プログラミング習得

#### 教育現場
- 誰の為に：プログラミング初学者
- 解決目的：プログラミング習得・エンジニア人口増加の一助

#### ライブラリとしての活用
- 誰の為に：
  - ハードウェア未経験者
  - ソフトウェアエンジニア
- 解決目的：
  - Web の技術でハードウェアを動かしたい
    - ブラウザ
    - JavaScript
    - Node.js 

#### 上記以外の何か
- 誰の為に：
  - TBD
- 解決目的：
  - TBD

### 課題
- エンジニアが楽しめるイベント企画
  - ミートアップ？もくもく会？
- リソース不足
  - サポート対象を限定する
    - ハード・ソフトのサポート対象を決める
    - サポート・メンテナンス対象
        - デバイスドライバー
        - GPIO/I2C
        - OS と ハードの組み合わせ

**Comment by tadfmac** [comment]
誰のために、何を解決するもの/コトを作るのか？という視点が大切と思います。
↑他の手段にない特徴が見出せれば、活動や開発の意味があると思います。

**Reply by gurezo** [reply]
@tadfmac 
加筆してみました。

cc @satakagi @kou029w  @sizuhiko @elie-j @dynamis 
皆さんのご意見も頂きたいです。

**Comment by gurezo** [comment]
議論が進まないので、一旦 close
